### PR TITLE
chore: add git actions 

### PR DESCRIPTION
**Pull request by tac0turtle**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Lint 
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
Comment (julienrbrt) [inline .github/workflows/lint.yml:6]: You may want to rename the master branch to main before merging this.

**Review by julienrbrt — APPROVED** [review]
lgtm, granted the master branch is renamed to main prior to merging.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ci-${{ github.ref }}-linting
+  cancel-in-progress: true
+
+jobs:
+  rust:
```
Comment (julienrbrt) [inline .github/workflows/lint.yml:16]: Do we want to run golangci-lint for the go part?

Comment (tac0turtle) [reply .github/workflows/lint.yml:16]: yes, wanted to add it in a different PR

**Pull request merged**